### PR TITLE
Refactoring/#267 switch codebuild waiter to tenacity

### LINKED ISSUE #254: Create AI-Lab Release 2.0.0
- https://github.com/exasol/ai-lab/issues/252
- https://github.com/exasol/ai-lab/issues/255
- https://github.com/exasol/ai-lab/issues/249
- https://github.com/exasol/notebook-connector/issues/89
- wait for ticket #247

Release process
- [x] Update version number in user guide etc.
- [x] Execute release process 
- [x] Update links in README.md to point to tags instead of latest state of branch `main`

### LINKED ISSUE #267: Refactor CodeBuildWaiter
Currently, timings only can be specified in terms of seconds.
Proposal: Use tenacity and benefit from support for arbitrary `timedelta` expressions.

### PR DESCRIPTION
**Pull request by ckunki**
Closes #267 
Closes #254

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Key Features:
 
 Feel free to explore the Exasol AI-Lab and unleash your data science potential!
 
-The AI-Lab is available in multiple [Editions](doc/user_guide/editions.md) involving different technology stacks, see also common [System Requirements](doc/user_guide/system-requirements.md).
+The AI-Lab is available in multiple [Editions](blob/2.0.0/doc/user_guide/editions.md) involving different technology stacks, see also common [System Requirements](doc/user_guide/system-requirements.md).
```
Comment (kaklakariada) [inline README.md:14]: This hard-codes the version. Is this intended? Maybe also update the paths to the other .md files.

File: exasol/ds/sandbox/lib/aws_access/waiter/codebuild_waiter.py [hunk 2]
```diff
@@ -10,42 +14,44 @@ class CodeBuildWaiter:
...
-    def _wait_for(seconds: int, interval: int) -> Iterable[int]:
-        for _ in range(int(seconds / interval)):
-            yield interval
-
-    def wait(self, timeout_in_seconds: int = 60*60*4, interval_in_seconds: int = 30):
+    def wait(
+            self,
+            timeout: timedelta = timedelta(hours=3, minutes=30),
+            interval: timedelta = timedelta(minutes=10),
+    ):
```
Comment (kaklakariada) [inline exasol/ds/sandbox/lib/aws_access/waiter/codebuild_waiter.py:26]: 👍

File: exasol/ds/sandbox/lib/aws_access/waiter/codebuild_waiter.py [hunk 3]
```diff
@@ -10,42 +14,44 @@ class CodeBuildWaiter:
...
+            return build_status
 
+        try:
+            status = poll_build_status()
+            if status in ['FAILED', 'FAULT', 'STOPPED', 'TIMED_OUT']:
+                raise RuntimeError(f"Build ({self._build_id}) failed with status: {status}")
+            if status != "SUCCEEDED":
+                raise RuntimeError(f"Build {self._build_id} has unknown build status: {status}")
+        except RetryError:
+            raise TimeoutError(f"Build {self._build_id} ran into timeout.")
```
Comment (kaklakariada) [inline exasol/ds/sandbox/lib/aws_access/waiter/codebuild_waiter.py:57]: 👍

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ckunki) [reply README.md:14]: The hard-coded link to tag 2.0.0 is deliberate. 
I updated the link to `jupyter.md`, too. 

The other links are lower priority, as pointing to developer guide and an image, whereas our major concern is users we want to point to the latest released version of the documentation. 

I also wanted to keep the impact small. Additionally, the links need to be excluded from link checker and cannot be resolved locally. So all in all, this is a poor-man's versioning. :slightly_frowning_face:

File: .github/workflows/check_ci.yaml [hunk 4]
```diff
@@ -80,6 +87,14 @@ jobs:
      - name: Setup Python & Poetry Environment
        uses: ./.github/actions/prepare_poetry_env
 
+     - name: Run Unit Tests
```
Comment (tkilias) [inline .github/workflows/check_ci.yaml:90]: create new job and make the integration and notebook tests depend on it with needs_to

File: .lycheeignore [hunk 5]
```diff
@@ -0,0 +1 @@
+blob/[0-9]+\.[0-9]+\.[0-9]+/doc/.*
```
Comment (tkilias) [inline .lycheeignore:1]: probably not needed, if you use absolute urls in the Readme

File: test/unit/aws/test_codebuild_waiter.py [hunk 6]
```diff
@@ -10,56 +11,58 @@
 BUILD_ID = 123
 
 
-def codebuild_client(build_duration: int, finish_status: str):
+def ms(value: int) -> timedelta:
+    return timedelta(milliseconds=value)
 
-    start_time = time.time()
 
+def codebuild_client(build_duration: timedelta, finish_status: str):
```
Comment (tkilias) [inline test/unit/aws/test_codebuild_waiter.py:18]:
```suggestion
def codebuild_client(build_duration: timedelta, finish_status: str) -> MagicMock:
```

File: .github/workflows/check_ci.yaml (hunk 4 above)
Comment (ckunki) [reply .github/workflows/check_ci.yaml]: See next push

File: .lycheeignore (hunk 5 above)
Comment (ckunki) [reply .lycheeignore]: See next push (deleted the file)

**Review by tkilias — APPROVED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**